interfaces/apparmor/backend_test.go

```diff
@@ -822,6 +822,10 @@ func (s *backendSuite) TestSetupSnapConfineGeneratedPolicyNoNFS(c *C) {
 	restore := apparmor.MockIsHomeUsingNFS(func() (bool, error) { return false, nil })
 	defer restore()
 
+	// Make it appear as if overlay was not used.
+	restore = apparmor.MockIsRootWritableOverlay(func() (string, error) { return "", nil })
```

Comment:
This looks fine! I wonder if we should just add this mocking as part of the backendSuite.SetUp() just once?

Reply:
I was looking at this, there are some parts that need the "other" behavior. I think that instead I would look at discarding all of the mocking (the problem with our mocking is that we never know we need it in advance unless tests fail) and passing an interface with those methods. Then tests always pass a fake implementation with appropriate answers and everything is explicit.